### PR TITLE
add ability to override build configs

### PR DESCRIPTION
**Pull request by iamJoeTaylor**
This add the ability to override some of the default build setup. I needed this so I could set react-dom as an external. Let me know what you think.

### CODE REVIEW

File: bin/config.js [hunk 1]
```diff
@@ -64,15 +90,29 @@ const config = (entry, out, bundleName, experimental) => {
   };
 
   // options for minified bundle extend from baseOptions
-  const minOptions = Object.assign({}, baseOptions, {
-    bundleName: `${libraryName}.min.js`,
-    minify: true,
-  });
+  const minOptions = Object.assign(
```
Comment (adrianmcli) [inline bin/config.js]: Object spread notation is now at Stage-3, I think we can use it to make this (and the next two `Object.assign` calls a bit cleaner.

File: bin/config.js [hunk 2]
```diff
@@ -51,7 +55,29 @@ const getConfig = ({ entry, out, libraryName, bundleName, minify, presets }) =>
...
+    if (!fs.existsSync(pkgLoc)) return {};
+  } catch (e) {
+    return {};
+  }
+
+  const packageJSON = require(pkgLoc);
+
+  if (!packageJSON['react-build-dist']) return {};
+
+  return packageJSON['react-build-dist'];
```
Comment (adrianmcli) [inline bin/config.js]: Consider refactoring lines 69 to 71 with:

```jsx
return packageJSON['react-build-dist'] || {};
```

File: bin/index.js [hunk 3]
```diff
@@ -11,13 +11,15 @@ const config = require('./config.js');
 program
   .version('0.0.1')
   .option('-b, --bundle-name [filename]', 'Output bundle filename')
+  .option('-p, --package-json [path]', 'path to package.json directory with react-build-dist config override')
```
Comment (adrianmcli) [inline bin/index.js]: Just for consistency's sake, maybe it's best to capitalize the `P` in the description:

```
'Path to package.json directory...
```

**Review by adrianmcli — CHANGES_REQUESTED** [review]
Thanks for taking the time to submit this PR! It's greatly appreciated. I only have a few minor change requests. If you could do that, I owe you a million github stars. 🌟

**Comment by iamJoeTaylor** [comment]
@adrianmcli done! 🌟

**Review by adrianmcli — APPROVED** [review]
LGTM

**Comment by iamJoeTaylor** [comment]
giphy 4

**Pull request merged**